tortoise/tortoise.go

```diff
@@ -256,41 +260,48 @@ func (t *turtle) getGoodBallot(logger log.Log) (types.BallotID, types.LayerID) {
 
 // firstDisagreement returns first layer where local opinion is different from ballot's opinion within sliding window.
 func (t *turtle) firstDisagreement(ctx *tcontext, blid types.LayerID, ballotID types.BallotID, disagreements map[types.BallotID]types.LayerID) (types.LayerID, error) {
```

Comment:
doesn't look like disagreement is used...

Reply:
can you elaborate? you mean this method is used in BaseBallot method

Reply:
i mean it's not used in this particular method `firstDisagreement`

Reply:
thanks, i accidentally dropped an optimization that if base ballot is not consistent then i can just reuse disagreement layer. will need to think of the benchmark for this case